ui/raidboss/emulator/data/PopupTextAnalysis.js

```diff
@@ -27,29 +75,37 @@ export default class PopupTextAnalysis extends StubbedPopupText {
       for (const trigger of this.triggers) {
         const r = log.match(trigger.localRegex);
         if (r) {
-          // Some async magic here, force waiting for the entirety of
-          // the trigger execution before continuing
-          const delayPromise = new Promise((res) => {
-            this.delayResolver = res;
+          const resolver = this.currentResolver = new Resolver();
+          resolver.status = {
```

Comment:
Confirming: this status lives on `Resolver` because some triggers are delayed and so you need to maintain the status of multiple triggers now? I'm mostly confused because the status seems unrelated to resolving (more like trigger helper??), so it's a bit of a mismatch at first glance.

Reply:
This is an awkward setup, because the way the analysis was written originally it was really cornered into handling only one trigger at a time. This is difficult because we need to be able to reference both the resolver and the status at various points and the scoping for the current trigger needs to change depending on which async function is currently executing.

As such, the `Resolver` and `status` properties need to be accessible async, and the only `PopupText` property we can piggyback off of to handle this is `triggerHelper`. So basically they both get references to each other and get stored on `triggerHelper` as well, so that we can dynamically change scope in the async executors.

The only other real way I can see to handle this without some other significant change to the way the base raidboss `PopupText` class works would be to have an instance of `PopupTextAnalysis` per trigger per player which is... excessive and slow.